### PR TITLE
travis-ci update: Add cppcheck run on codebase

### PR DESCRIPTION
**Pull request by csravelar**
This patch adds a cppcheck install and execution command to
run cppcheck on each pull request to make sure that there is
no resulting errors. Addresses issue #357 

Jira: None
Test: Travis fails if cppcheck finds errors and passes otherwise.
Signed-off-by: Kelly Ledford kelly.ledford@intel.com
Signed-off-by: Richard Avelar richard.avelar@intel.com

### CODE REVIEW

**Comment by kalyankondapally** [comment]
"/home/travis/.travis/job_stages: line 57: travisci/cpp_check_patch.sh: Permission denied"

**Comment by kalyankondapally** [comment]
@stripes416 I would prefer fixing all issues as part of this commit and enforcing these checks need to pass going forward for any PR.

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +14,8 @@ install:
 - wget https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-linux.zip -O /tmp/ninja.zip
 - sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.9 50
 - sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.9 50
+- sudo apt-get install cppcheck/trusty-backports
+- sudo apt-get install cppcheck
```
Comment (harishkrupo) [inline .travis.yml]: redundant install?

File: .travis.yml [hunk 2]
```diff
@@ -26,6 +29,9 @@ script:
 - tar -xvf travisci/resources/hwc-install.tar.xz -C $HWC_BUILD_DIR
 - mkdir -p $WLD/share/aclocal
 
+# Run cppcheck on codebase
+- cppcheck --quiet --force --error-exitcode=1 .
```
Comment (harishkrupo) [inline .travis.yml]: Maybe add this to travisci/check_patch.sh? 
We could later add this to the post commit hook to ensure that this is checked while committing.

File: travisci/check_patch.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from subprocess import Popen, PIPE, STDOUT
```
Comment (harishkrupo) [inline travisci/check_patch.py]: Shebang: #!/usr/bin/python3 (to ensure it is run under python3 interpreter)

Comment (csravelar) [reply travisci/check_patch.py]: Ahh, this should save me from adding python3 in "python3 travisci/check_patch.py", thanks!

**Comment by kalyankondapally** [comment]
merged

**Pull request closed without merging**